src/main/java/org/zalando/fahrschein/NakadiReader.java

```diff
@@ -192,11 +193,11 @@ private void readEvent(final Iterator<T> source, final List<T> target) throws Js
         }
     }
 
-    public void run() throws IOException {
+    public void run() throws IOException, BackoffException {
```

Comment:
What about catching the backoff exceptions in here and then only expose the `IOException`?